packages/bull-vault/test/fuzz-test/LeverageBullFuzzTest.t.sol

```diff
@@ -0,0 +1,244 @@
+        uint256 eulerUsdcBalance = IERC20(usdc).balanceOf(euler);
+
+        if (usdcToBorrow > eulerUsdcBalance) usdcToBorrow = eulerUsdcBalance;
+
+        uint256 usdcBalanceBefore = IERC20(usdc).balanceOf(auction);
+
+        vm.startPrank(auction);
+        leverageBull.depositAndBorrowFromLeverage(0, usdcToBorrow);
+        vm.stopPrank();
+        
+        uint256 usdcBalanceAfter = IERC20(usdc).balanceOf(auction);
+
+        assertEq(usdcBalanceAfter.sub(usdcBalanceBefore), usdcToBorrow);
+
+        vm.startPrank(auction);
+        uint256 usdcToRepay = IEulerDToken(dToken).balanceOf(auction);
+        IERC20(usdc).approve(address(leverageBull), usdcToRepay);
+        leverageBull.auctionRepayAndWithdrawFromLeverage(usdcToRepay, 0);
+
+        assertEq(IEulerDToken(dToken).balanceOf(auction), 0);
+        assertEq(usdcBalanceAfter.sub(IERC20(usdc).balanceOf(auction)), 0);
+    }
+
+    function testFuzzingBorrowAndRepayPartialBalanceOf(
+        uint256 _wethToDeposit,
+        uint256 _percentage
+    ) public {
+        _wethToDeposit = bound(_wethToDeposit, 1e18, 100000000e18);
+        _percentage = bound(_percentage, 0, 100);
+
+        vm.deal(auction, _wethToDeposit);
+
+        vm.startPrank(auction);
+        // deposit first WETH in euler
+        IWETH9(weth).deposit{value: _wethToDeposit}();
+        IWETH9(weth).approve(address(leverageBull), _wethToDeposit);
+        leverageBull.depositAndBorrowFromLeverage(_wethToDeposit, 0);
+        vm.stopPrank();
+
+        vm.prank(address(leverageBull));
+        IEulerMarkets(eulerMarketsModule).enterMarket(0, weth);
+
+        uint256 balanceOfUnderlying =
+            IEulerEToken(eToken).balanceOfUnderlying(address(leverageBull));
+
+        uint256 ethUsdPrice = UniOracle._getTwap(ethUsdcPool, weth, usdc, TWAP, false);
+        uint256 usdcToBorrow = _wethToDeposit.wmul(ethUsdPrice).wdiv(TARGET_CR).div(WETH_DECIMALS_DIFF);
+        uint256 eulerUsdcBalance = IERC20(usdc).balanceOf(euler);
+
+        if (usdcToBorrow > eulerUsdcBalance) usdcToBorrow = eulerUsdcBalance;
+
+        uint256 usdcBalanceBefore = IERC20(usdc).balanceOf(auction);
+
+        vm.startPrank(auction);
+        leverageBull.depositAndBorrowFromLeverage(0, usdcToBorrow);
+        vm.stopPrank();
+        
+        uint256 usdcBalanceAfter = IERC20(usdc).balanceOf(auction);
+
+        assertEq(usdcBalanceAfter.sub(usdcBalanceBefore), usdcToBorrow);
+
+        vm.startPrank(auction);
+        uint256 debtAmount = IEulerDToken(dToken).balanceOf(auction);
+        uint256 usdcToRepay = debtAmount.mul(_percentage).div(100);
+        IERC20(usdc).approve(address(leverageBull), usdcToRepay);
+        leverageBull.auctionRepayAndWithdrawFromLeverage(usdcToRepay, 0);
+
+        assertEq(IEulerDToken(dToken).balanceOf(auction).sub(debtAmount), usdcToRepay);
+        assertEq(usdcBalanceAfter.sub(IERC20(usdc).balanceOf(auction)), usdcToRepay);
```

Comment:
this seems to be right, but very different from the above assert, not sure why (the one that says change is 0)

Reply:
usdcToRepay is getting set to 0 per above debtAmount issue